src/typechecker.erl

```diff
@@ -2462,7 +2462,10 @@ do_type_check_expr_in(Env, ResTy, {record, Anno, Name, Fields} = Record) ->
             {VarBinds, Cs2} = type_check_fields(Env, Rec, Fields),
             {VarBinds, constraints:combine(Cs1, Cs2)};
         {fields_tys, Tyss, Cs1} ->
-            case type_check_record_union_in(Env, Tyss, Fields) of
+            Tyss2 = lists:map(fun (?type(any)) -> get_record_fields(Name, Anno, Env);
+                                  (Other) -> Other
+                              end, Tyss),
+            case type_check_record_union_in(Env, Tyss2, Fields) of
```

Comment:
Why not add a clause in `type_check_record_union_in` instead?

```Erlang
type_check_record_union_in(Env, [?type(any) | Tyss], Fields) ->
    ... ;
```

Reply:
We don't have `Name` and `Anno` there, so calling `get_record_fields` just seemed simpler where I did it. Not a big difference, though.

Reply:
OK